lib/RGBot.js

```diff
@@ -583,196 +1098,360 @@ const RGBot = class {
                 await this.#bot.equip(bestHarvestTool, 'hand');
                 return this.#bot.heldItem;
             } catch (err) {
-                console.error('Unable to equip a better tool', err);
+                this.#logWarn('Error trying to equip a better tool', err);
                 return null;
             }
         }
     }
 
+    /**
+     * Finds the best harvest tool in the bot inventory for mining the specified block.
+     * If we don't have the best tool, also checks if dig time is infinite because it can't be harvested without a tool
+     *
+     * @param block {Block}
+     * @returns {BestHarvestTool}
+     */
+    bestHarvestTool(block) {
+        const availableTools = this.#bot.inventory.items()
+        const effects = this.#bot.entity.effects
+
+        let fastest = Infinity
+        let bestTool = null
+        for (const tool of availableTools) {
+            const enchants = (tool && tool.nbt) ? nbt.simplify(tool.nbt).Enchantments : []
+            const digTime = block.digTime(tool ? tool.type : null, false, false, false, enchants, effects)
+            if (digTime < fastest) {
+                fastest = digTime
+                bestTool = tool
+            }
+        }
+
+        return new BestHarvestTool(bestTool, fastest)
+    }
+
+    /**
+     * <i><b>Experimental</b></i>
+     *
+     * This finds the most powerful melee attack item in the bot inventory
+     *
+     * Note: Today this only prioritizes weapon type, but does not prioritize weapon rarity/enchantments/etc
+     *
+     * @returns {Item|undefined}
+     */
+    bestAttackItemMelee() {
+        return this.#bot.inventory.items().filter((item) => {
+            let itemName = item.name.toLowerCase()
+            return itemName.includes("sword") || itemName.includes("trident") || itemName.endsWith("_axe") || itemName.endsWith("_pickaxe");
+        }).sort((a, b) => {
+            if (a.name.endsWith("_axe") && b.name.endsWith("_pickaxe")) {
+                return -1
+            } else if (a.name.includes("trident") && b.name.endsWith("axe")) {
+                return -1
+            } else if (a.name.includes("sword")) {
+                return -1
+            } else if (a.name === b.name) {
+                return 0
+            } else {
+                return 1
+            }
+        }).shift()
+    }
+
     /**
      * Dig the given Block.
-     * This will equip the most appropriate tool in the Bot's inventory for this Block type.
+     * This will equip the most appropriate tool in the bot inventory for this Block type.
      * @param {Block} block The Block instance to dig
      * @returns {Promise<boolean>} Whether the Block was successfully dug
      */
     async digBlock(block) {
```

Comment:
Can we add a comment to the doc about what happens if the player is not in range of the block?